### ISSUE TITLE
Doxygen does not recognize using declaration on base class constructors and demands description of the return value.

### ISSUE THREAD

**Issue by rlalik** [doxygen/doxygen#7736]
When having following block of code:
~~~c++
class A {
public:
    /// constructor
    A() {}
};
class B : public A {
public:
    using A::A;
};
~~~

But this code after running doxygen yields warning:

    warning: return type of member B::A is not documented

Doxygen should recognize constructors and suppress the warning.

**Comment by albert-github** [comment]
With the limited information I wasn't able to reproduce the problem.

- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.
- Please also specify the doxygen version used.

**Comment by rlalik** [comment]
Doxygen 1.8.16
Example is attached.

[example_using.tar.gz](https://github.com/doxygen/doxygen/files/4554545/example_using.tar.gz)

**Comment by albert-github** [comment]
With version 1.8.16 I see the warning as indicated. With version 1.8.17 and 1.8.18 )the current version) this message is gone.

I think the problem has been solved in the mean time.

**Comment by rlalik** [comment]
Indeed, version bump helped. But I found then another issue with void functions with custom prefixes, e.g. for attributes. The example is attached. Probbaly it should be reported as separate bug. I did not notice the problem with functions returning non-void values. Doxy version 1.8.18.
[void_functions_prefix.zip](https://github.com/doxygen/doxygen/files/4575713/void_functions_prefix.zip)


**TLDR**

    #define SYM_EXPORT __attribute__((visibility("default")))
    SYM_EXPORT void foo();
    __attribute__((visibility("default"))) void bar();

yields

    class.h:13: warning: Member __attribute__((visibility("default"))) void bar() (function) of class A is not documented.
    class.h:12: warning: return type of member A::foo is not documented

**Comment by albert-github** [comment]
I think in this case there are  a few issues:
- `bar` is really not documented
- the `__attribute__` is unknown to doxygen, it should be `PREDEFINED` as well (see http://doxygen.nl/manual/preprocessing.html, here also `__attrubute__` is mentioned)
- `EXPAND_ONLY_PREDEF     = YES` is set in the Doxyfile, so `SVM_EXPORT` is not expanded, so either set `http://doxygen.nl/manual/preprocessing.html=NO` or predefine `SVM_EXPORT` as well.

**Comment by rlalik** [comment]
OK, following your instructions, I made it working. Since it worked in 1.8.16, I did not expected this to be configuration issue in 1.8.18, and 1.8.17 (I did not change the configuration) but rather regression.

According to `bar`, in the example indeed I forgot to comment it but in the real code I had problem the same like with foo.

I think I can close the issue now.
